include/dgl/aten/macro.h

```diff
@@ -171,6 +171,24 @@
   }                                                           \
 } while (0)
 
+#define ATEN_FLOAT_BOOL_TYPE_SWITCH(val, DType, val_name, ...) do {     \
+  if ((val).code == kDGLInt && (val).bits == 8) {              \
+    typedef int8_t DType;                                     \
+    {__VA_ARGS__}                                             \
+  } else if ((val).code == kDGLUInt && (val).bits == 8) {      \
+    typedef uint8_t DType;                                    \
+    {__VA_ARGS__}                                             \
+  } else if ((val).code == kDGLFloat && (val).bits == 32) {    \
+    typedef float DType;                                      \
+    {__VA_ARGS__}                                             \
+  } else if ((val).code == kDGLFloat && (val).bits == 64) {    \
+    typedef double DType;                                     \
+    {__VA_ARGS__}                                             \
+  } else {                                                    \
+    LOG(FATAL) << (val_name) << " can only be int8, uint8, float32 or float64"; \
+  }                                                           \
+} while (0)
```

Comment:
Two earlier comments from @jermainewang 

> Please add some docstring about how to use this macro.

> The code doesn't look like related to bool type at all.

Reply:
I renamed it to `ATEN_FLOAT_INT8_UINT8`.  Previously I named it `BOOL` because `int8` and `uint8` arrays store boolean arrays in current PyTorch/TF/etc.